test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -88,3 +94,221 @@ def test_invalid_script_out(self):
+        self.server.restart()
+
+        # Setup node
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a,
+                            id=self.mom.shortname, expect=True)
+
+        # Submit job
+        a = {'Resource_List.foobar': '4'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # Job must run successfully
+        a = {'job_state': 'R', 'Resource_List.foobar': 4}
+        self.server.expect(JOB, a, id=jid)
+
+    def test_res_long_neg(self):
+        """
+        Test that server_dyn_res accepts command line arguments to the
+        commands it runs. Resource value set to a negative long int.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        # Create a resource of type long
+        resname = "foobar"
+        attr = {"type": "long"}
+        self.server.manager(MGR_CMD_CREATE, RSC, attr, id=resname, expect=True)
+
+        # Add resource to sched_config's 'resources' line
+        self.scheduler.add_resource(resname)
+
+        # Add server_dyn_res entry in sched_config
+        self.scheduler.set_sched_config(
+            {'server_dyn_res': '"foobar !/bin/echo -1"'})
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+
+        # Restart the server so the resourcedef info will be updated
+        self.server.restart()
+
+        # Setup node
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a,
+                            id=self.mom.shortname, expect=True)
+
+        # Submit job
+        a = {'Resource_List.foobar': '1'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # Check for the expected log message for insufficient resources
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foobar (R: 1 A: -1 T: -1)"
+
+        # The job shouldn't run
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_whitespace(self):
+        """
+        Test for parse errors when more than one white space
+        is added between the resource name and the !<script> in a
+        server_dyn_res line. There shouldn't be any errors.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        # Create a resource of type long
+        resname = "foo"
+        attr = {"type": "string"}
+        self.server.manager(MGR_CMD_CREATE, RSC, attr, id=resname, expect=True)
+
+        # Add resource to sched_config's 'resources' line
+        self.scheduler.add_resource(resname)
+
+        # Prep for server_dyn_resource scripts. Script "get_foo"
+        # generates file "got_foo" and returns 1.
+        script_body = "echo get_foo > /tmp/got_foo; echo 1"
+
+        fpath_in = os.path.join(os.sep, "tmp", "get_foo")
```

Comment:
use du.mkstemp for creating temporary files. this comment applies everywhere.

Reply:
Done